Implemented collateral updates for QAT deletes.

yonatan-dp opened the pull request:
Implemented collateral updates for QAT deletes.

-Perform Collateral EM Updates
-Perform Collateral QAT Updates

yonatan-dp committed:
#
Implemented collateral updates for QAT deletes.

-Perform Collateral EM Updates
-Perform Collateral QAT Updates

yonatan-dp committed:
Merge branch 'develop' into fix/#6902-delete-qat-collateral-updates

djw4erg commented on src/test-summary-workspace/test-summary.service.ts:
```diff
@@ -836,6 +836,10 @@ export class TestSummaryWorkspaceService {
+
+        //Finally, perform the updates (reset needs eval flag, etc) for those records
+        // that may have been collaterally affected by this change.
+        await this.updateCollaterallyAffectedRecords(testSumId);
```
The underlying SP uses the TEST_SUMMARY and QA_SUPP_DATA rows that are deleted earlier in this method.  It looks like the rows would have been pull from Official if they exist there, but if the deleted test has never been submitted, I think the lack of TEST_SUMMARY and QA_SUPP_DATA rows would prevent the collateral updates from occurring.

yonatan-dp replied:
Yep. I have updated so that the call happens before the deletions occur. The queries occur within a transaction so it is still all or nothing.

maxdiebold-erg replied:
Should `updateCollaterallyAffectedRecords` use that transaction? Currently it does not. For an example, see how the method is updated [here](https://github.com/US-EPA-CAMD/easey-qa-certification-api/blob/963ce66cb3a5857b8b46c42c273b0b77a0fe418e/src/test-summary-workspace/test-summary.service.ts#L916) for the broader transaction context.

yonatan-dp replied:
@maxdiebold-erg You are correct, it should. But, that update to the method to use transactions is sitting unmerged in this PR:  `https://github.com/US-EPA-CAMD/easey-qa-certification-api/pull/750`.  It only exists in `fix/#6559-prevent-partial-QA-imports` branch.  So, all of the `updateCollaterallyAffectedRecords` calls (and there are a few) have to be updated similarly - and will be when that PR is merged. I don't know if it makes sense to only update one `updateCollaterallyAffectedRecords` signature that will be overriden by the merge.   To kep this functionality consistent, I have updated it to use the same transaction.

maxdiebold-erg replied:
Sure, but this is a different transaction context, and I wouldn't consider this PR "complete" without the change, so I do think it makes sense to update the single method in this PR. I agree it won't matter once the other PR is merged, though.

yonatan-dp replied:
@maxdiebold-erg I already updated it and pushed (last sentence in my last comment)

maxdiebold-erg replied:
Thanks!

yonatan-dp committed:
#
Implemented collateral updates for QAT deletes.

-Perform Collateral EM Updates
-Perform Collateral QAT Updates

yonatan-dp committed:
#
Implemented collateral updates for QAT deletes.

-Perform Collateral EM Updates
-Perform Collateral QAT Updates

yonatan-dp committed:
Merged from Develop

yonatan-dp committed:
Merge branch 'develop' into fix/#6902-delete-qat-collateral-updates

djw4erg commented on src/test-summary-workspace/test-summary.service.ts:
```diff
@@ -845,6 +845,11 @@ export class TestSummaryWorkspaceService {
+      // First, perform the updates (reset needs eval flag, etc) for those records
+      // that may have been collaterally affected by this change.
+      // This must be done BEFORE deletion so the procedures can access the test data.
+      await this.updateCollaterallyAffectedRecords(testSumId, transactionalEntityManager);
```
I believe that this needs to happen before the Test Summary delete at line 846.  Am I misreading this?  Sorry that I did not ask about this previously.

yonatan-dp committed:
Corrected merge issue

mark-hayward-erg committed:
Merge branch 'develop' into fix/#6902-delete-qat-collateral-updates

mark-hayward-erg committed:
Merge branch 'develop' into fix/#6902-delete-qat-collateral-updates

mark-hayward-erg committed:
Merge branch 'develop' into fix/#6902-delete-qat-collateral-updates
